workspaces/services/guacamole-azure-dsvm/terraform/dsvm.tf

```diff
@@ -0,0 +1,128 @@
+resource "azurerm_network_interface" "internal" {
+  name                = "internal-nic-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+
+  ip_configuration {
+    name                          = "primary"
+    subnet_id                     = data.azurerm_subnet.dsvms.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_security_group" "dsvmnsg" {
+  name                = "nsg-${local.service_resource_name_suffix}"
+  location            = data.azurerm_resource_group.ws.location
+  resource_group_name = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "allow-rdp-inbound-from-vnet-dsvm" {
+  name                        = "RDP"
+  priority                    = 1010
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = 3389
+  source_address_prefix       = "VirtualNetwork"
+  destination_address_prefix  = "*"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  resource_group_name         = data.azurerm_resource_group.ws.name
+}
+
+resource "azurerm_network_security_rule" "deny-all-inbound-override-dsvm" {
+  access                      = "Deny"
+  destination_address_prefix  = "*"
+  destination_port_range      = "*"
+  direction                   = "Inbound"
+  name                        = "deny-inbound-override"
+  network_security_group_name = azurerm_network_security_group.dsvmnsg.name
+  priority                    = 900
```

Comment:
Lower number = higher priority, isn't it?
Doesn't this blocks the rule allowing access through port 3389?

Reply:
Thanks. good catch.